backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/ContractsDAO.java

```diff
@@ -197,4 +197,20 @@ public I_C_Flatrate_Term retrieveLatestFlatrateTermForBPartnerId(@NonNull final
 				.create()
 				.first();
 	}
+
+	@Cached(cacheName = I_C_Flatrate_Term.Table_Name + "#by#BPartnerId")
+	@Override
+	public I_C_Flatrate_Term retrieveFirstFlatrateTermForBPartnerId(@NonNull final BPartnerId bpartnerId)
+	{
+		return Services.get(IQueryBL.class).createQueryBuilder(I_C_Flatrate_Term.class)
+				.addOnlyActiveRecordsFilter()
+				.addOnlyContextClient()
+				.addEqualsFilter(I_C_Flatrate_Term.COLUMNNAME_Bill_BPartner_ID, bpartnerId.getRepoId())
```

Comment:
```suggestion
				.addEqualsFilter(I_C_Flatrate_Term.COLUMNNAME_Bill_BPartner_ID, bpartnerId)
```
the filter supports RepoIdAwares directly